File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} list1
     * @param {ListNode} list2
     * @return {ListNode}
     */
    mergeTwoLists(list1, list2) {
        let result=null,head=null
        if(list1===null){
            return list2
        }
        if(list2===null){
            return list1
        }
        if(list1.val<=list2.val){
            result=list1
            list1=list1.next
        }else{
            result=list2
            list2=list2.next
        }
        head=result
        while(list1!==null && list2!==null){
            if(list1.val<=list2.val){
                result.next=list1
                list1=list1.next
                result=result.next
            }else{
                result.next=list2
                list2=list2.next
                result=result.next
            }
        }
        if(list1==null){
            result.next=list2
        }else{
            result.next=list1
        }
        return head
    }
}

File: Data Structures & Algorithms/reorder-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        slow,fast=head,head.next
        while fast and fast.next:
          slow=slow.next
          fast=fast.next.next
        l2=slow.next
        slow.next=None
        l1=head
        current=l2
        previous=None
        while current:
          temp=current.next
          current.next=previous
          previous=current
          current=temp
        l2=previous
        count=0
        while l2:
          temp=l1.next
          temp2=l2.next
          l1.next=l2
          l2.next=temp
          l1=temp
          l2=temp2